src/data/actions/courseInfo.js

```diff
@@ -68,6 +69,26 @@ function editCourseFail(error) {
   return { type: EDIT_COURSE_FAIL, error };
 }
 
+function clearCourseSaved() {
+  return { type: CLEAR_COURSE_SAVED };
+}
+
+function updateFrontendAfterSave(change, currentValues, initialImageSrc, initialCourseRuns) {
+  return (dispatch) => {
+    // We need to overwrite these values because they are changed in the backend so the
+    // frontend does not have the updated values by default. This will allow the form to
+    // return to being pristine after saving.
+    change('imageSrc', initialImageSrc);
+    for (let i = 0; i < initialCourseRuns.length; i += 1) {
+      // eslint-disable-next-line no-param-reassign
+      currentValues.course_runs[i].status = initialCourseRuns[i].status;
+    }
+    // Need to reset courseInfo.courseSaved to false so we don't continuously
+    // overwrite the currentValues.
+    dispatch(clearCourseSaved());
+  };
+}
```

Comment:
Will talk in person about all this stuff. I just need some context. Looks gross on the surface, but probably better than all our other options.

Reply:
We need to overwrite the current form values because they are changed in the backend so frontend does not have the updated values by default. This will allow the form to return to being pristine after saving.